kedro/config/omegaconf_config.py

```diff
@@ -445,6 +441,28 @@ def _resolve_environment_variables(config: dict[str, Any]) -> None:
         else:
             OmegaConf.resolve(config)
 
+    @staticmethod
+    def _destructive_merge(config, env_config, env_path):
+        # Destructively merge the two env dirs. The chosen env will override base.
+        common_keys = config.keys() & env_config.keys()
+        if common_keys:
+            sorted_keys = ", ".join(sorted(common_keys))
+            msg = (
+                "Config from path '%s' will override the following "
+                "existing top-level config keys: %s"
+            )
+            _config_logger.debug(msg, env_path, sorted_keys)
+
+        config.update(env_config)
+        return config
+
+    @staticmethod
+    def _soft_merge(config, env_config):
+        config = OmegaConf.to_container(
+            OmegaConf.merge(config, env_config), resolve=True
+        )
+        return config
```

Comment:
```suggestion
    def _soft_merge(config, env_config):
        config = OmegaConf.merge(config, env_config)
        return config
```
Is the to_container necessary? It should merge two resolved configuration here and we don't need to resolve it again

Reply:
@ankatiyar I update my suggestion. Sorry I mean we shouldn't resolve this again. The resulted configuration from individual environment is a pure dictonary.

![image](https://github.com/kedro-org/kedro/assets/18221871/57c5329b-16d0-48e6-bb89-d6e5c5411851)